File: src/demo/z_rotation_sine_wave.cpp
// MIT License
// 
// Copyright (c) 2020 Jean-Louis Haywood
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

#include "z_rotation_sine_wave.hpp"

void OmnificDemo::ZRotationSineWave::on_entity_start()
{
    this->time = 0.0;
    this->transform = Omnific::CPPEntityContext::get_entity()->get_transform();
}

void OmnificDemo::ZRotationSineWave::on_fixed_update()
{
    float delta = Omnific::CPPEntityContext::get_time_delta();
    float z_rotation_speed = 1.0;
    float amplitude = 45.0;
    this->transform->rotation.z = amplitude * sin(this->time * z_rotation_speed);
    this->time += delta;
}

File: src/applications/demo/app_data/tests/scripts/python/z_rotation_sine_wave.py
import omnific
import math

class omnific_script:

    def __init__(self):
        self.time = 0
        pass

    def on_entity_start(self):
        self.transform = omnific.get_transform()
        pass

    def on_fixed_update(self):
        delta = omnific.get_time_delta()
        z_rotation_speed = 1.0
        amplitude = 45.0
        
        self.transform.set_z_rotation(amplitude * math.sin(self.time * z_rotation_speed))
        self.time += delta
        pass
